src/Compilers/Core/Portable/CommandLine/AnalyzerConfigSet.cs

```diff
@@ -349,5 +411,157 @@ internal static bool TryParseSeverity(string value, out ReportDiagnostic severit
             severity = default;
             return false;
         }
+
+        internal static bool IsPartialGlobalConfig(AnalyzerConfig config) => config.GlobalSection.Properties.ContainsKey(GlobalKey);
+
+        /// <summary>
+        /// Merge any partial global configs into a single global config, and remove the partial configs
+        /// </summary>
+        /// <param name="analyzerConfigs">An <see cref="ArrayBuilder{T}"/> of <see cref="AnalyzerConfig"/> containing a mix of regular and unmerged partial global configs</param>
+        /// <param name="diagnostics">Diagnostics produced during merge will be added to this bag</param>
+        /// <returns>A <see cref="GlobalAnalyzerConfig" /> that contains the merged partial configs, or <c>null</c> if there were no partial configs</returns>
+        internal static GlobalAnalyzerConfig? MergeGlobalConfigs(ArrayBuilder<AnalyzerConfig> analyzerConfigs, out ImmutableArray<Diagnostic> diagnostics)
```

Comment:
`private`? Do we use this anywhere else?

Reply:
It's used pretty heavily in the tests

---
In reply to: [423392254](https://github.com/dotnet/roslyn/pull/43889#discussion_r423392254) [](ancestors = 423392254)